### ISSUE TITLE
[BUG/ISSUE] High memory usage

### ISSUE THREAD

**Issue by mboop127** [marticliment/ElevenClock#1078]
When left running for a while, elevenclock eventually balloons to 1-3 mb of ram. I saw a similar bug reported earlier but it was marked resolved.

I do regularly leave programs running (python scripts etc) overnight that cause windows explorer to bug out, so it's possible that elevenclock is just reacting to that, but I would appreciate it if you could investigate.

**Comment by marticliment** [comment]
do you mean 1-3gb, right?


has this issue happened again?
if so, can you please try resetting ElevenClock?

**Comment by mboop127** [comment]
yes gb sorry. It happens eventually after every program or system restart.

**Comment by marticliment** [comment]
can you please share your log after this issue happening?

**Comment by marticliment** [comment]
Hi, could you please try the new beta? 
https://github.com/marticliment/ElevenClock/releases/tag/4.0.0-beta1

**Comment by mboop127** [comment]
![image](https://user-images.githubusercontent.com/26417201/210881923-fd4ab99f-c1b1-4131-bb9a-96c1d41a5e7c.png)
here is a screenshot - could you walk me through sending you the log? or do you just want me to try the beta?

**Comment by marticliment** [comment]
so, please first install the beta.

if the issue persists, please:
 1. without quitting ElevenClock, open the settings window
 2. Scroll down to "debugging info" and click on that section
 3. A pop-up will show up. Then choose the "Open log" entry
 4. Export the log as a text file and attach it here.

Thanks!

**Comment by mboop127** [comment]
The update did not resolve the issue, was using 1.1gb of ram when I opened settings to export the log, though once I opened settings usage went back to 100mb or so. Log attached
[log.txt](https://github.com/marticliment/ElevenClock/files/10366538/log.txt)

**Comment by marticliment** [comment]
On your user home directlry there's a folder called .elevenclock.

Could you please compres it and send it back to me?

**Comment by mboop127** [comment]
attached
[.elevenclock.zip](https://github.com/marticliment/ElevenClock/files/10368706/default.elevenclock.zip)

**Comment by hoekbrwr** [comment]
It looks like there are major memory leaks in the program. Memory usage starts with 65MB and then keeps on rising with every action you do, like reload clocks. If you regularly sync the time on the internet this can reach  1 GB in a day easily. 
Have a look at the structure to identify these leaks. Google for the general causes and have an aerial view of your program's structure. It seems to be at a high level in the program because lots of actions make the memory usage rise. Good luck with that! 
I however like the program with its many settings. I now know why there is a menu point "reload the program" lol

**Comment by marticliment** [comment]
I am aware of this rises, but I have never experienced those crazy levels of ram. 
The issue is still being investigated, but under normal usage without mnessing with the settings, the ram should not rise at all

**Comment by marticliment** [comment]
Next version of ElevenClock includes a new set of improvements that should prevent this issue.
